update Miri

RalfJung opened the pull request:
r? @thomcc for the lib changes (removing a `cfg(miri)` that is no longer needed)

RalfJung committed:
clean up retagging recursive traversal a bit

klensy committed:
remove unused shell_escape dep

RalfJung committed:
Auto merge of #2670 - RalfJung:retag, r=RalfJung

recreated by hand

RalfJung committed:
Auto merge of #2671 - klensy:shell_escape, r=RalfJung

recreated by hand

RalfJung committed:
update josh scripts
- support pulling specific commit
- make rust-version update a separate commit to avoid confusing josh
- after pushing, check that we have a clear round-trip

RalfJung committed:
update rust-version

RalfJung committed:
empty commit to go through bors

bors committed:
Auto merge of #2676 - RalfJung:empty, r=RalfJung

empty commit to go through bors

Go through bors once to clean up after the force push, have CI run, all the usual.

RalfJung committed:
remove CI badge

bors committed:
Auto merge of #2677 - RalfJung:badge, r=RalfJung

remove CI badge

We don't actually run CI on the `master` branch (because bors checks things before they can even be on master), so this doesn't really show anything. rustc doesn't have a badge either. I think we can live without it. :D

RalfJung committed:
Preparing for merge from rustc

RalfJung committed:
Merge from rustc

RalfJung committed:
implement clock_gettime on macos

bors committed:
Auto merge of #2679 - RalfJung:clock_gettime, r=RalfJung

implement clock_gettime on macos

and pull in rustc changes so we can test this against https://github.com/rust-lang/rust/pull/103594.

Fixes https://github.com/rust-lang/miri/issues/2664

RalfJung committed:
cfg(miri) no longer needed in sys/unix/time.rs

rust-log-analyzer commented:
The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/3505551869/jobs/5871926778) [(plain)](https://github.com/rust-lang/rust/commit/975f74d26675d813096dae03b3f3e6f481efb9e2/checks/9595510540/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
  IMAGE: x86_64-gnu-tools
##[endgroup]
From https://github.com/rust-lang/rust
 * branch              master     -> FETCH_HEAD
Searching for toolstate changes between c5d82ed7a4ad94a538bb87e5016e7d5ce0bd434b and 7d9c6e8ae6d959b64d81770143ba9ea9b7801530
Tool subtrees were updated
##[group]Run src/ci/scripts/verify-channel.sh
src/ci/scripts/verify-channel.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
---
+ /tmp/checktools.sh ../x.py
Building rustbuild
    Finished dev [unoptimized] target(s) in 0.05s
Building stage0 std artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
error: the lock file /checkout/Cargo.lock needs to be updated but --locked was passed to prevent this
If you want to try to generate the lock file without accessing the network, remove the --locked flag and use --offline instead.
cat: /tmp/toolstate/toolstates.json: No such file or directory
```

</details>

thomcc commented:
r=me on lib changes, which are trivial.

RalfJung committed:
Preparing for merge from rustc

RalfJung committed:
Merge from rustc

RalfJung committed:
fix unused warning in a test

bors committed:
Auto merge of #2680 - RalfJung:rustup, r=RalfJung

Rustup

RalfJung committed:
update lockfile

RalfJung commented:
@bors r+ p=1

bors commented:
:pushpin: Commit f70fd7853c9a0134997c37987c11b275ff9c16e0 has been approved by `RalfJung`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=RalfJung f70fd7853c9a0134997c37987c11b275ff9c16e0 -->
<!-- homu: {"type":"Approved","sha":"f70fd7853c9a0134997c37987c11b275ff9c16e0","approver":"RalfJung","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

RalfJung committed:
make miri-seed a regular integer, and also set layout-seed in many-seeds

bors committed:
Auto merge of #2681 - RalfJung:seed, r=oli-obk

make miri-seed a regular integer, and also set layout-seed in many-seeds

This makes the seed format consistent between `-Zlayout-seed` and `-Zmiri-seed`.

RalfJung commented:
@bors r-
I have an upcoming Miri PR that should be included

RalfJung committed:
make align_offset always work on no-provenance pointers

bors committed:
Auto merge of #2683 - RalfJung:align_offset, r=RalfJung

make align_offset always work on no-provenance pointers

Fixes https://github.com/rust-lang/miri/issues/2682

RalfJung commented:
@bors r+ p=1

bors commented:
:pushpin: Commit c043a0e7d6f347f9b2c7f08c5b3a179470e0f0c5 has been approved by `RalfJung`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=RalfJung c043a0e7d6f347f9b2c7f08c5b3a179470e0f0c5 -->
<!-- homu: {"type":"Approved","sha":"c043a0e7d6f347f9b2c7f08c5b3a179470e0f0c5","approver":"RalfJung","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit c043a0e7d6f347f9b2c7f08c5b3a179470e0f0c5 with merge 911cbf8e46f0f3b044c87fc5cbca922878db757b...
<!-- homu: {"type":"BuildStarted","head_sha":"c043a0e7d6f347f9b2c7f08c5b3a179470e0f0c5","merge_sha":"911cbf8e46f0f3b044c87fc5cbca922878db757b"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/3508076477/jobs/5877355347)
Approved by: RalfJung
Pushing 911cbf8e46f0f3b044c87fc5cbca922878db757b to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"RalfJung","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/3508076477/jobs/5877355347"},"merge_sha":"911cbf8e46f0f3b044c87fc5cbca922878db757b"} -->

rust-timer commented:
Finished benchmarking commit (911cbf8e46f0f3b044c87fc5cbca922878db757b): [comparison URL](https://perf.rust-lang.org/compare.html?start=9cdfe03b0601d5328406746a7923a8a4eaa0cf3c&end=911cbf8e46f0f3b044c87fc5cbca922878db757b&stat=instructions:u).

### Overall result: ✅ improvements - no action needed


@rustbot label: -perf-regression



### [Instruction count](https://perf.rust-lang.org/compare.html?start=9cdfe03b0601d5328406746a7923a8a4eaa0cf3c&end=911cbf8e46f0f3b044c87fc5cbca922878db757b&stat=instructions:u)
This is a highly reliable metric that was used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -     | -              | 0     |
| Regressions ❌ <br /> (secondary)  | -     | -              | 0     |
| Improvements ✅ <br /> (primary)   | -0.7% | [-0.9%, -0.5%] | 7     |
| Improvements ✅ <br /> (secondary) | -     | -              | 0     |
| All ❌✅ (primary)                 | -0.7% | [-0.9%, -0.5%] | 7     |

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=9cdfe03b0601d5328406746a7923a8a4eaa0cf3c&end=911cbf8e46f0f3b044c87fc5cbca922878db757b&stat=max-rss)
<details>
<summary>Results</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean | range        | count |
|:----------------------------------:|:----:|:------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -    | -            | 0     |
| Regressions ❌ <br /> (secondary)  | 2.2% | [2.2%, 2.2%] | 1     |
| Improvements ✅ <br /> (primary)   | -    | -            | 0     |
| Improvements ✅ <br /> (secondary) | -    | -            | 0     |
| All ❌✅ (primary)                 | -    | -            | 0     |
</details>

### [Cycles](https://perf.rust-lang.org/compare.html?start=9cdfe03b0601d5328406746a7923a8a4eaa0cf3c&end=911cbf8e46f0f3b044c87fc5cbca922878db757b&stat=cycles:u)
This benchmark run did not return any relevant results for this metric.

lqd commented:
Note for weekly triage: `bitmaps` is currently noisy
